src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -39,6 +46,24 @@ public void deleteCart(@PathVariable Long cartId) {
         service.deleteCart(cartId);
     }
 
+    @GetMapping("/item/{itemId}")
+    public ItemDto getItem(@PathVariable Long itemId) {
+        return mapper.mapToItemDto(service.getItem(itemId));
+    }
 
+    @PostMapping(path ="/newItem", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void addItem(@RequestBody ItemDto itemDto){
+        service.addItem(mapper.mapToItem(itemDto));
+    }
+
+    @DeleteMapping("/item/{itemId}")
+    public void deleteItem(@PathVariable("itemId") Long itemId){
+        service.deleteItem(itemId);
+    }
+
+    @PostMapping(path ="/newOrder", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public OrderDto createAnOrder(OrderDto orderDto) {
```

Comment:
Brak @RequestBody -> godzinę szukałem.
W UserOrder jest coś takiego @MapsId -> jak się przyjrzysz w bazie z powodu tego nie tworzy kolumny id. To jest do usunięcia, ten zapis źle generuje tabelę w Hibernate. Mamy relacje 1;1 jednostronną.
We wszystkich encjach (cart, item, order, user) dodać strategy = GenereatedValue.Identity

Reply:
Tak jak napisałam na czacie, myslę że artykuł godny zastosowania: https://vladmihalcea.com/change-one-to-one-primary-key-column-jpa-hibernate/

Reply:
rzeczywistość (testy circleci) zweryfikowały moje plany... :D Patrz commit: resolving circleci error